lib/bundle/brew_installer.rb

```diff
@@ -14,21 +14,35 @@ def initialize(name, options = {})
       @full_name = name
       @name = name.split("/").last
       @args = options.fetch(:args, []).map { |arg| "--#{arg}" }
+      @unlink = options.fetch(:unlink, false)
+      @stop_service = options.fetch(:stop_service, false)
       @restart_service = options.fetch(:restart_service, false)
     end
 
     def run
-      install_or_upgrade and restart_service!
+      install_change_state! && service_change_state!
     end
 
-    def install_or_upgrade
+    def unlink?
+      @unlink
+    end
+
+    def install_change_state!
       if installed?
-        upgrade!
+        if unlink?
+          unlink!
+        else
+          upgrade!
+        end
       else
         install!
       end
     end
```

Comment:
I don't think we should use `brew "foo", unlink: :true` as DSL to handle what really is conflicting list. The reasons are (a) `brew "foo"` as a DSL is used to declare a dependency, and this is not. (b) it will interfere with the functions of `check` and `cleanup`. 

As I'm still thinking it's formula's duty to declare its conflicted formulae, I acknowledge this info can be missing. So how about a `brew "foo", conflicts_with: %w[a b]` to supply missing information. Therefore, rest of things can be handled as unlink/stop if conflict.